numpy/lib/tests/test_arraypad.py

```diff
@@ -831,19 +825,25 @@ def test_check_03(self):
         b = np.array([1, 2, 3, 2, 1, 2, 3, 2, 1, 2, 3])
         assert_array_equal(a, b)
 
-    def test_check_padding_an_empty_array(self):
-        a = np.pad(np.zeros((0, 3)), ((0,), (1,)), mode='reflect')
-        b = np.zeros((0, 5))
-        assert_array_equal(a, b)
 
-    def test_padding_empty_dimension(self):
-        match = "There aren't any elements to reflect in axis 0"
+class TestEmptyArray(object):
+    """Check how padding behaves on arrays with an empty dimension."""
+
+    @pytest.mark.parametrize("mode", _all_modes.keys() - {"constant", "empty"})
+    def test_pad_empty_dimension(self, mode):
```

Comment:
Hmmm, the difference seems to be due to the order of tests/parameterization. I am not sure that is an actual issue, and I am a bit confused as to why it should only show up here, since I would think the dictionary keys are always random order. My guess is that adding a `sorted(...)` would fix it, I am not sure that is nice though.

Reply:
Please change the parametrization on the test(s) to have a sorted order, yes. The Shippable tests are running on more than 1 core & the parallel orchestration mechanism used by xdist doesn't like the lack of ordering.

There are a bunch of pytest-xdist issues floating around about this, and maybe someday pytest-xdist will be able to handle this, but not yet.

Reply:
@lagru, I realize that it only was an issue here, but to me it seems like this should also occur for the other uses of `_all_modes`, so maybe we should make `_all_modes` a `collection.OrderedDict`, or is that one fine @tylerjereddy? I am a bit surprised this doesn't occur elsewhere, but...

Reply:
@seberg That's indeed strange. On the danger of saying something stupid or obvious: the shippable configuration seems to use Python 3.7 [which guarantees that dictionaries are ordered](https://mail.python.org/pipermail/python-dev/2017-December/151283.html) (and likely their key-views as well?). If that assumption is correct, not using an OrderedDict would only be an issue for a CI-setup that uses an older Python version (3.5, I think 3.6 was ordered as well but not yet official) together with pytest-xdist...

Reply:
Ah, might be going back and forth with python versions whether or not it works, so maybe if 3.7 is fine and we are using 3.7 we are unlikely to go to an older version which might create issues.

Reply:
`_all_modes.keys()` returns a `dict_keys` object (view) which allows set operations. Subtracting the set `{"constant", "empty"}` seems to produce a `set` object. My guess is that because `set` is not an ordered type, `pytest-xdist`detects the inconsistency for the parallell workflow.

The dictionary was probably already "sorted" according to insertion order since it was tested in Python 3.7, so it is really the sorting of the `set` object (and its conversion to a `list` which retains the sorting) that likely solves the issue here.

That's probably also why the other `_all_modes` uses are just fine -- those are just iterators that preserve the original dictionary insertion order.

Of course, there have been arguments that I shouldn't have used `xdist` in the CI--if it becomes a big issue we can turn off parallel tests. I think this has been done for some platforms / CI infrastructures, but has mostly worked quite well on Shippable for some time now, so maybe this set parametrization thing is rare enough that we can live with converting to a sorted list in the those rare cases where we do that.

Reply:
Sounds like this is all good with the `sorted` in this one place then.